mailauth/contrib/user/models.py

```diff
@@ -68,6 +86,19 @@ def has_usable_password(self):
     class Meta(AbstractUser.Meta):
         abstract = True
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._email = self.email
+
+    def save(self, *args, **kwargs):
+        if self.email and ((self._email != self.email) or not self.pk):
+            self.email_hash = hashlib.md5(self.email.lower().encode()).digest()
```

Comment:
Since you are testing for `hexdigest` in your tests, this bit is untested I guess.

Reply:
changing my phrase here as well.
Your tests do test for `digest` but the code uses hexdigest in other places.

Reply:
...and the md5 function on the migration also does hexdigest as a small comment showed.